modules/standard/IO.chpl

```diff
@@ -7936,8 +7938,8 @@ proc fileWriter.writeBinary(b: bytes, size: int = b.size) throws {
                        due to a :ref:`system error<io-general-sys-error>`.
 */
 proc fileWriter.writeBinary(const ref data: [?d] ?t, param endian:ioendian = ioendian.native) throws
-  where (d.rank == 1 && d.stridable == false && !d.isSparse()) && (
-          isIntegralType(t) || isRealType(t) || isImagType(t) || isComplexType(t))
+  where d.rank == 1 && d.isRectangular() && d.hasUnitStride() && (
```

Comment:
Do these methods not work with associative? If so, can you open an issue to request that we document that somewhere and send it to the IO subteam?

Reply:
Each method where I added “isRectangular()” has the following in its chpldoc: “Note that this routine currently requires a 1D rectangular non-strided array.” So I did not add anything new, I just made the where-clauses look the same.